pydrive2/fs/spec.py

```diff
@@ -542,6 +542,34 @@ def cp_file(self, lpath, rpath, **kwargs):
             buffer = io.BytesIO(stream.read())
             self.upload_fobj(buffer, rpath)
 
+    def mv(self, path1, path2, recursive=False, maxdepth=None, **kwargs):
+
+        if recursive:
+            raise NotImplementedError("Recursive move is not supported")
+        if maxdepth:
+            raise NotImplementedError("Max depth move is not supported")
+
+        file1_name = posixpath.basename(path1)
+        file2_name = posixpath.basename(path2)
+
+        file1_parent = self._parent(path1)
+        file2_parent = self._parent(path2)
+
+        file1_id = self._get_item_id(path1)
+
+        file1 = self.client.CreateFile({"id": file1_id})
+
+        file1.FetchMetadata(fields="title,parents")
+
+        if file1_name != file2_name:
+            file1["title"] = file2_name
+
+        if file1_parent != file2_parent:
+            file2_parent_id = self._get_item_id(file2_parent)
+            file1["parents"] = [{"id": file2_parent_id}]
+
+        file1._FilesUpdate()
```

Comment:
after this operation is done we need to update or invalidate the ids cache `self._ids_cache` (see around how it's managed, probably we might need another helper that is wrapped with `@wrap_with(threading.RLock())`)

Reply:
sorry but i couldn't find any instance where the cache gets invalidated. 
even in `rm_file` the cache is not touched:

https://github.com/iterative/PyDrive2/blob/64a50a1580b2d276f8c9563f5ba620f7e7706119/pydrive2/fs/spec.py#L590-L605

how do i remove an element from the cache?

Reply:
resolving this since the whole cache mechanism is quite broken in GDrive's fsspec. Let's add as a TODO here